src/main/java/Event.java

```diff
@@ -0,0 +1,34 @@
+public class Event extends Task {
+    private static final String TYPE = "E";
+    private final String from;
+    private final String to;
+
+    Event(String input, String from, String to) {
+        super(input, TYPE);
+        this.from = from;
+        this.to = to;
+    }
+
+    Event(String input, boolean status, String from, String to) {
+        super(input, TYPE, status);
+        this.from = from;
+        this.to = to;
+    }
+
+    @Override
+    String encodeString() {
+        return String.format(("%s|%s|%s|%s|%s"), super.getType(),
+                super.getStatus() ? "X" : " ",
+                super.getInput(), this.from, this.to);
```

Comment:
This line is very complicated, especially with the conditional operators inside. Try to simplify this expression

Reply:
Do check for other complicated expressions in your code as well